libnavigation-copilot/src/main/java/com/mapbox/navigation/copilot/HistoryEvents.kt

```diff
@@ -0,0 +1,64 @@
+package com.mapbox.navigation.copilot
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+
+internal interface EventDTO
+
+/**
+ * To create a new event, extend the [HistoryEvent] and provide a unique
+ * name in snake case.
+ *
+ * @property snakeCaseEventName unique name in snake case
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+sealed class HistoryEvent(internal val snakeCaseEventName: String, internal val eventDTO: EventDTO)
+
+internal const val SEARCH_RESULTS_EVENT_NAME = "search_results"
+internal const val SEARCH_RESULT_USED_EVENT_NAME = "search_result_used"
+internal const val INIT_ROUTE_EVENT_NAME = "init_route"
+internal const val DRIVE_ENDS_EVENT_NAME = "drive_ends"
+internal const val GOING_TO_FOREGROUND_EVENT_NAME = "going_to_foreground"
+internal const val GOING_TO_BACKGROUND_EVENT_NAME = "going_to_background"
+internal const val NAV_FEEDBACK_SUBMITTED_EVENT_NAME = "nav_feedback_submitted"
+
+/**
+ * SearchResultsEvent
+ *
+ * @property searchResults
+ *
+ * @see [HistoryEvent]
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+data class SearchResultsEvent(val searchResults: SearchResults) :
+    HistoryEvent(SEARCH_RESULTS_EVENT_NAME, searchResults)
+
+/**
+ * SearchResultUsedEvent
+ *
+ * @property searchResultUsed
+ *
+ * @see [HistoryEvent]
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+data class SearchResultUsedEvent(val searchResultUsed: SearchResultUsed) :
```

Comment:
This was to avoid the need of opt-in `ExperimentalPreviewMapboxNavigationAPI` in some other internal classes (e.g. any class that relies on `MapboxCopilot.pushStatusObservers`.

Reply:
2 main reasons:
- We cannot generate the events internally as we don't integrate the Search SDK as part of the Nav SDK (yet)
- We also have that open for other search providers